diff -Nur a/dracut-functions.sh b/dracut-functions.sh
--- a/dracut-functions.sh	2015-11-25 13:22:28.000000000 +0000
+++ b/dracut-functions.sh	2018-01-10 06:32:03.255256206 +0000
@@ -652,7 +652,7 @@
     if [[ "$(get_cpu_vendor)" == "AMD" ]]; then
         # If family greater or equal than 0x15
         if [[ $family -ge 21 ]]; then
-            printf "microcode_amd_fam15h.bin"
+            printf "microcode_amd_fam%xh.bin" $family
         else
             printf "microcode_amd.bin"
         fi
