diff --git a/src/backend/f_generic.sh b/src/backend/f_generic.sh
index d692c46..ce413fe 100644
--- a/src/backend/f_generic.sh
+++ b/src/backend/f_generic.sh
@@ -38,7 +38,7 @@ checkjailsum () {
 	fi
 }
 
-jaildkmsbuild () {
+dkmsmodbuild () {
 	checkifroot
 	if [[ -x /usr/sbin/dkms ]] ; then
 		for i in $(dkms status | cut -d " " -f1,2 | sed -e 's/,//g' | sed -e 's/ /\//g' | sed -e 's/://g') ; do
diff --git a/src/backend/f_makeiso.sh b/src/backend/f_makeiso.sh
index be5a550..8cadcfa 100644
--- a/src/backend/f_makeiso.sh
+++ b/src/backend/f_makeiso.sh
@@ -56,7 +56,7 @@ jailisostart () {
 mkliveimg () {
 	# create live filesystem image layout
 	mkdir -p "$jailsynctarget"
-	truncate -s 10G ""$jailsynctarget".img"
+	dd if=/dev/zero of=""$jailsynctarget".img" bs=1M count=15360
 	sync
 	mkfs.ext2 -F ""$jailsynctarget".img"
 	mkdir -p "$jailsyncsource"
diff --git a/src/frontend/cli/vasile.sh b/src/frontend/cli/vasile.sh
index aeb27b3..557caee 100755
--- a/src/frontend/cli/vasile.sh
+++ b/src/frontend/cli/vasile.sh
@@ -34,7 +34,7 @@ case $1 in
 		srcmode
 		;;
 	--dkms)
-		dkmsmod
+		dkmsmodbuild
 		;;
 	--help)
 		showhelp
