diff --git a/src/backend/f_modeswitch.sh b/src/backend/f_modeswitch.sh
index 0f66964..323afba 100644
--- a/src/backend/f_modeswitch.sh
+++ b/src/backend/f_modeswitch.sh
@@ -101,7 +101,7 @@ injectportfulltree () {
 setbinmodecfg () {
 	ln -sf "$jailportcfgsource" "$jailportcfgtarget"
 	ln -sf "$jailportcfgtarget"/make.conf.amd64-binmode "$jailportcfgtarget"/make.conf
-	eselect profile set 1
+	eselect profile set redcore:default/linux/amd64/13.0
 	env-update
 	. /etc/profile
 }
@@ -115,7 +115,7 @@ binmode () {
 setmixedmodecfg () {
 	ln -sf "$jailportcfgsource" "$jailportcfgtarget"
 	ln -sf "$jailportcfgtarget"/make.conf.amd64-mixedmode "$jailportcfgtarget"/make.conf
-	eselect profile set 1
+	eselect profile set redcore:default/linux/amd64/13.0
 	env-update
 	. /etc/profile
 }
@@ -129,7 +129,7 @@ mixedmode () {
 setsrcmodecfg () {
 	ln -sf "$jailportcfgsource" "$jailportcfgtarget"
 	ln -sf "$jailportcfgtarget"/make.conf.amd64-srcmode "$jailportcfgtarget"/make.conf
-	eselect profile set 1
+	eselect profile set redcore:default/linux/amd64/13.0
 	env-update
 	. /etc/profile
 }
diff --git a/src/backend/v_jail.sh b/src/backend/v_jail.sh
index 11e9945..39dfcf8 100644
--- a/src/backend/v_jail.sh
+++ b/src/backend/v_jail.sh
@@ -4,7 +4,7 @@ export local distupstream="gentoo"
 export local distname="redcore"
 export local disttarget="desktop"
 export local distver="1"
-export local kernver="4.14.10-"$distname"-lts"
+export local kernver="4.14.12-"$distname"-lts"
 
 export local jailx64=""$distname"_"$distver"_core_x64.squashfs"
 export local jailx64sum=""$jailx64".md5sum"
