diff -Nur a/kernel/sched/MuQSS.c b/kernel/sched/MuQSS.c
--- a/kernel/sched/MuQSS.c	2019-01-05 22:51:24.547448624 +0000
+++ b/kernel/sched/MuQSS.c	2019-01-05 22:58:29.821451056 +0000
@@ -1021,6 +1021,10 @@
 #define CPUIDLE_THREAD_BUSY	(16)
 #define CPUIDLE_DIFF_NODE	(32)
 
+#ifdef CONFIG_SCHED_SMT
+DEFINE_STATIC_KEY_FALSE(sched_smt_present);
+#endif
+
 /*
  * The best idle CPU is chosen according to the CPUIDLE ranking above where the
  * lowest value would give the most suitable CPU to schedule p onto next. The
