diff -Nur a/src/VBox/Additions/linux/drm/Makefile.module.kms b/src/VBox/Additions/linux/drm/Makefile.module.kms
--- a/src/VBox/Additions/linux/drm/Makefile.module.kms	2017-07-17 16:58:30.000000000 +0100
+++ b/src/VBox/Additions/linux/drm/Makefile.module.kms	2017-07-22 14:29:44.977047225 +0100
@@ -50,7 +50,7 @@
 ifneq ($(KERN_VERSION),24)
 MOD_CFLAGS += -include $(MANGLING)
 endif
-MOD_INCL   = $(addprefix -I$(KBUILD_EXTMOD),/ /include)
+MOD_INCL   = $(addprefix -I$(KBUILD_EXTMOD),/ /include /r0drv/linux)
 # What on earth is this?
 MOD_INCL  += $(addprefix -I$(KBUILD_EXTMOD)/vboxvideo,/ /include)
 MOD_INCL  += -Iinclude/drm
