diff --git a/src/backend/libsisyphus.py b/src/backend/libsisyphus.py
index 069d563..e83258a 100755
--- a/src/backend/libsisyphus.py
+++ b/src/backend/libsisyphus.py
@@ -168,6 +168,10 @@ def startSearch(pkgList):
 def startUpdate():
     syncAll()
 
+def startSync():
+    syncGitRepos()
+    syncPortageCfg()
+
 def sysInfo():
     subprocess.check_call(['emerge', '--info'])
 
diff --git a/src/frontend/cli/sisyphus-cli.py b/src/frontend/cli/sisyphus-cli.py
index 965e3f1..9d347ab 100755
--- a/src/frontend/cli/sisyphus-cli.py
+++ b/src/frontend/cli/sisyphus-cli.py
@@ -18,7 +18,7 @@
         elif "remove-orphans" in sys.argv[1:]:
             removeOrphans()
         elif "update" in sys.argv[1:]:
-            startUpdate()
+            startSync()
         elif "upgrade" in sys.argv[1:]:
             startUpgrade()
         elif "search" in sys.argv[1:]:
