diff -Nur a/dkms.conf b/dkms.conf
--- a/dkms.conf	2017-07-18 12:50:58.000000000 +0100
+++ b/dkms.conf	2017-07-23 14:40:45.746169005 +0100
@@ -14,18 +14,18 @@
 # hope that it will be useful, but WITHOUT ANY WARRANTY of any kind.
 #
 
-PACKAGE_NAME="vboxguest"
-PACKAGE_VERSION=5.1.24_OSE
+PACKAGE_NAME="virtualbox-guest-dkms"
+PACKAGE_VERSION=5.1.24
 AUTOINSTALL=yes
 
 BUILT_MODULE_NAME[0]="vboxguest"
 BUILT_MODULE_LOCATION[0]="vboxguest"
-DEST_MODULE_LOCATION[0]="/kernel/misc"
+DEST_MODULE_LOCATION[0]="/extra/dkms"
 
 BUILT_MODULE_NAME[1]="vboxsf"
 BUILT_MODULE_LOCATION[1]="vboxsf"
-DEST_MODULE_LOCATION[1]="/kernel/misc"
+DEST_MODULE_LOCATION[1]="/extra/dkms"
 
 BUILT_MODULE_NAME[2]="vboxvideo"
 BUILT_MODULE_LOCATION[2]="vboxvideo"
-DEST_MODULE_LOCATION[2]="/kernel/misc"
+DEST_MODULE_LOCATION[2]="/extra/dkms"
