diff -Nur a/dkms.conf b/dkms.conf
--- a/dkms.conf	2017-12-19 21:31:10.000000000 +0000
+++ b/dkms.conf	2017-12-22 02:51:53.824679071 +0000
@@ -15,7 +15,7 @@
 #
 
 PACKAGE_NAME="vboxguest"
-PACKAGE_VERSION=5.2.4_OSE
+PACKAGE_VERSION=5.2.4
 AUTOINSTALL=yes
 
 BUILT_MODULE_NAME[0]="vboxguest"
@@ -25,7 +25,3 @@
 BUILT_MODULE_NAME[1]="vboxsf"
 BUILT_MODULE_LOCATION[1]="vboxsf"
 DEST_MODULE_LOCATION[1]="/kernel/misc"
-
-BUILT_MODULE_NAME[2]="vboxvideo"
-BUILT_MODULE_LOCATION[2]="vboxvideo"
-DEST_MODULE_LOCATION[2]="/kernel/misc"
