diff -Nur a/pcmanfm/mainwindow.cpp b/pcmanfm/mainwindow.cpp
--- a/pcmanfm/mainwindow.cpp	2019-02-24 00:16:13.000000000 +0000
+++ b/pcmanfm/mainwindow.cpp	2019-06-01 12:31:10.004634642 +0100
@@ -201,7 +201,7 @@
 
     // use generic icons for view actions only if theme icons don't exist
     ui.actionIconView->setIcon(QIcon::fromTheme(QLatin1String("view-list-icons"), style()->standardIcon(QStyle::SP_FileDialogContentsView)));
-    ui.actionThumbnailView->setIcon(QIcon::fromTheme(QLatin1String("dialog-information"), style()->standardIcon(QStyle::SP_FileDialogInfoView)));
+    ui.actionThumbnailView->setIcon(QIcon::fromTheme(QLatin1String("view-list-thumbnails"), style()->standardIcon(QStyle::SP_FileDialogInfoView)));
     ui.actionCompactView->setIcon(QIcon::fromTheme(QLatin1String("view-list-text"), style()->standardIcon(QStyle::SP_FileDialogListView)));
     ui.actionDetailedList->setIcon(QIcon::fromTheme(QLatin1String("view-list-details"), style()->standardIcon(QStyle::SP_FileDialogDetailedView)));
 
