diff -Nur a/pcmanfm/mainwindow.cpp b/pcmanfm/mainwindow.cpp
--- a/pcmanfm/mainwindow.cpp	2018-05-21 20:14:54.000000000 +0100
+++ b/pcmanfm/mainwindow.cpp	2019-02-02 12:12:29.726816339 +0000
@@ -164,10 +164,10 @@
     loadBookmarksMenu();
 
     // set generic icons for view actions
-    ui.actionIconView->setIcon(style()->standardIcon(QStyle::SP_FileDialogContentsView));
-    ui.actionThumbnailView->setIcon(style()->standardIcon(QStyle::SP_FileDialogInfoView));
-    ui.actionCompactView->setIcon(style()->standardIcon(QStyle::SP_FileDialogListView));
-    ui.actionDetailedList->setIcon(style()->standardIcon(QStyle::SP_FileDialogDetailedView));
+    ui.actionIconView->setIcon(QIcon::fromTheme(QLatin1String("view-list-icons"), style()->standardIcon(QStyle::SP_FileDialogContentsView)));
+    ui.actionThumbnailView->setIcon(QIcon::fromTheme(QLatin1String("view-list-thumbnails"), style()->standardIcon(QStyle::SP_FileDialogInfoView)));
+    ui.actionCompactView->setIcon(QIcon::fromTheme(QLatin1String("view-list-text"), style()->standardIcon(QStyle::SP_FileDialogListView)));
+    ui.actionDetailedList->setIcon(QIcon::fromTheme(QLatin1String("view-list-details"), style()->standardIcon(QStyle::SP_FileDialogDetailedView)));
 
     // Fix the menu groups which is not done by Qt designer
     // To my suprise, this was supported in Qt designer 3 :-(
