diff -Nur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2017-10-21 20:44:07.000000000 +0100
+++ b/CMakeLists.txt	2017-12-03 22:20:14.845402570 +0000
@@ -24,8 +24,8 @@
     find_package(Qt5DBus)
 endif()
 find_package(QTermWidget5 REQUIRED)
-find_package(lxqt-build-tools ${LXQTBT_MINIMUM_VERSION} REQUIRED)
-include(LXQtTranslateTs)
+#find_package(lxqt-build-tools ${LXQTBT_MINIMUM_VERSION} REQUIRED)
+#include(LXQtTranslateTs)
 include(LXQtCompilerSettings NO_POLICY_SCOPE)
 message(STATUS "Qt version: ${Qt5Core_VERSION}")
 
@@ -109,23 +109,23 @@
 qt5_wrap_ui( QTERM_UI ${QTERM_UI_SRC} )
 qt5_wrap_cpp( QTERM_MOC ${QTERM_MOC_SRC} )
 qt5_add_resources( QTERM_RCC ${QTERM_RCC_SRC} )
-lxqt_translate_ts(QTERM_QM
-    UPDATE_TRANSLATIONS
-        ${UPDATE_TRANSLATIONS}
-    SOURCES
-        ${QTERM_SRC}
-        ${QTERM_UI_SRC}
-        ${QTERM_MOC_SRC}
-    TRANSLATION_DIR "src/translations"
-    PULL_TRANSLATIONS
-        ${PULL_TRANSLATIONS}
-    CLEAN_TRANSLATIONS
-        ${CLEAN_TRANSLATIONS}
-    TRANSLATIONS_REPO
-        ${TRANSLATIONS_REPO}
-    TRANSLATIONS_REFSPEC
-        ${TRANSLATIONS_REFSPEC}
-)
+#lxqt_translate_ts(QTERM_QM
+#    UPDATE_TRANSLATIONS
+#        ${UPDATE_TRANSLATIONS}
+#    SOURCES
+#        ${QTERM_SRC}
+#        ${QTERM_UI_SRC}
+#        ${QTERM_MOC_SRC}
+#    TRANSLATION_DIR "src/translations"
+#    PULL_TRANSLATIONS
+#        ${PULL_TRANSLATIONS}
+#    CLEAN_TRANSLATIONS
+#        ${CLEAN_TRANSLATIONS}
+#    TRANSLATIONS_REPO
+#        ${TRANSLATIONS_REPO}
+#    TRANSLATIONS_REFSPEC
+#        ${TRANSLATIONS_REFSPEC}
+#)
 
 include_directories(
     "${PROJECT_SOURCE_DIR}"
