diff -burN usr.old/bin/skypeforlinux usr/bin/skypeforlinux
--- a/usr.old/bin/skypeforlinux	2017-12-09 03:43:34.598533060 +0200
+++ b/usr/bin/skypeforlinux	2017-12-09 03:48:47.582759089 +0200
@@ -3,10 +3,24 @@
 SCRIPT=$(readlink -f "$0")
 USR_DIRECTORY=$(readlink -f $(dirname $SCRIPT)/..)
 
+export HTTP_PROXY="${HTTP_PROXY:-}"
+export http_proxy="${http_proxy:-}"
+
+PROXY_SERVICE=""
+[ -n "${HTTP_PROXY}" ] && { PROXY_SERVICE="${HTTP_PROXY}"; }
+[ -n "${http_proxy}" ] && { PROXY_SERVICE="${http_proxy}"; }
+
+PROXY_SERVICE_ARGUMENT=""
+[ -n "${PROXY_SERVICE}" ] && { PROXY_SERVICE_ARGUMENT=" HTTP_PROXY=${PROXY_SERVICE} HTTPS_PROXY=${PROXY_SERVICE} http_proxy=${PROXY_SERVICE} https_proxy=${PROXY_SERVICE} "; }
+
 SKYPE_PATH=/opt/skypeforlinux/skypeforlinux
-SKYPE_LOGS="$HOME/.config/skypeforlinux/logs"
+SKYPE_LOGS="${HOME}/.config/skypeforlinux/logs"
+
+if [[ ! -e "${SKYPE_LOGS}" ]] ; then
+	mkdir -p "${SKYPE_LOGS}"
+fi
 
-mkdir -p $SKYPE_LOGS
+export ${PROXY_SERVICE_ARGUMENT}
 
-nohup "$SKYPE_PATH" --executed-from="$(pwd)" --pid=$$ "$@" > "$SKYPE_LOGS/skype-startup.log" 2>&1 &
+nohup "${SKYPE_PATH}" --executed-from="$(pwd)" --pid=$$ "$@" > "${SKYPE_LOGS}/skype-startup.log" 2>&1 &
 
