--- a/pindialog.ui	2016-09-28 14:42:34.630880165 -0700
+++ b/pindialog.ui	2016-09-28 16:11:21.815784765 -0700
@@ -54,7 +54,7 @@
   </property>
   <property name="windowIcon">
    <iconset>
-    <normaloff>../icons/scalable_bluemoon.svg</normaloff>../icons/scalable_bluemoon.svg</iconset>
+    <normaloff>/usr/share/pixmaps/blue_moon.svg</normaloff>/usr/share/pixmaps/blue_moon.svg</iconset>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
--- a/progress.ui	2016-09-28 14:42:34.630880165 -0700
+++ b/progress.ui	2016-09-28 16:15:08.245780710 -0700
@@ -53,7 +53,7 @@
   </property>
   <property name="windowIcon">
    <iconset>
-    <normaloff>../icons/scalable_bluemoon.svg</normaloff>../icons/scalable_bluemoon.svg</iconset>
+    <normaloff>/usr/share/pixmaps/blue_moon.svg</normaloff>/usr/share/pixmaps/blue_moon.svg</iconset>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
--- a/remoteselector.ui	2016-09-28 14:42:34.630880165 -0700
+++ b/remoteselector.ui	2016-09-28 16:46:23.665747125 -0700
@@ -30,7 +30,7 @@
   </property>
   <property name="windowIcon">
    <iconset>
-    <normaloff>../icons/scalable_bluemoon.svg</normaloff>../icons/scalable_bluemoon.svg</iconset>
+    <normaloff>/usr/share/pixmaps/blue_moon.svg</normaloff>/usr/share/pixmaps/blue_moon.svg</iconset>
   </property>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
--- a/remoteselector.cpp	2016-09-28 14:42:34.630880165 -0700
+++ b/remoteselector.cpp	2016-09-28 16:47:07.085746348 -0700
@@ -459,7 +459,7 @@
 
     trayIcon = new QSystemTrayIcon(this);
     trayIcon->setContextMenu(trayIconMenu);
-    trayIcon->setIcon(QIcon("../icons/scalable_bluemoon.svg"));
+    trayIcon->setIcon(QIcon("/usr/share/pixmaps/blue_moon.svg"));
 }
 void RemoteSelector::changeHostMode()
 {
