diff -Nur a/setup.py b/setup.py
--- a/setup.py	2017-03-09 03:38:06.000000000 +0000
+++ b/setup.py	2017-03-12 23:16:27.093692814 +0000
@@ -11,7 +11,7 @@
     # os.system('pyrcc5 -o meteo_qt/qrc_resources.py meteo_qt/resources.qrc')
     os.system('pylupdate5 meteo_qt/meteo_qt.pro')
     for ts in glob.glob('meteo_qt/translations/*.ts'):
-        os.system('lrelease {0} -qm {1}'.format(ts, (ts[:-2]+'qm')))
+        os.system('/usr/lib64/qt5/bin/lrelease {0} -qm {1}'.format(ts, (ts[:-2]+'qm')))
 
 
 setup(
