diff -Nur a/sopprotocol.cpp b/sopprotocol.cpp
--- a/sopprotocol.cpp	2023-04-29 18:43:36.000000000 +0100
+++ b/sopprotocol.cpp	2024-02-09 16:14:43.390296277 +0000
@@ -52,7 +52,7 @@
 
 ///////////////////////////////////////////////////////////////////////////
 
-void CSopProtocol::SetProtoPort(uint32_t port)
+void CSopProtocol::SetProtoPort(wxInt32 port)
 {
    m_protoport = port;
 };
@@ -66,7 +66,7 @@
 
 ///////////////////////////////////////////////////////////////////////////
 
-void CSopProtocol::SetPlayerPort(uint32_t port)
+void CSopProtocol::SetPlayerPort(wxInt32 port)
 {
    m_playerport = port;
 };
diff -Nur a/sopprotocol.h b/sopprotocol.h
--- a/sopprotocol.h	2023-04-29 18:43:36.000000000 +0100
+++ b/sopprotocol.h	2024-02-09 16:14:32.763295637 +0000
@@ -31,15 +31,15 @@
    void SetCmd(wxString cmd);
 
    int GetProtoport();
-   void SetProtoPort(uint32_t port);
+   void SetProtoPort(wxInt32 port);
 
    int GetPlayerPort();
-   void SetPlayerPort(uint32_t port);
+   void SetPlayerPort(wxInt32 port);
 
 
 private:
-   uint32_t m_playerport;
-   uint32_t m_protoport;
+   wxInt32 m_playerport;
+   wxInt32 m_protoport;
    long m_protopid;
    int SearchSop(wxString &cmd);
 
