diff -Nur a/src/modules/fstab/main.py b/src/modules/fstab/main.py
--- a/src/modules/fstab/main.py	2016-11-25 17:38:49.000000000 +0000
+++ b/src/modules/fstab/main.py	2017-06-02 15:46:38.124863165 +0100
@@ -257,12 +257,20 @@
                         check=check,
                         )
 
-        return dict(device="UUID=" + partition["uuid"],
-                    mount_point=mount_point or "swap",
-                    fs=filesystem,
-                    options=options,
-                    check=check,
-                    )
+        if "luksMapperName" in partition:
+            return dict(device="/dev/mapper/" + partition["luksMapperName"],
+                        mount_point=mount_point or "swap",
+                        fs=filesystem,
+                        options=options,
+                        check=check,
+                        )
+        else:
+            return dict(device="UUID=" + partition["uuid"],
+                        mount_point=mount_point or "swap",
+                        fs=filesystem,
+                        options=options,
+                        check=check,
+                        )
 
     def print_fstab_line(self, dct, file=None):
         """ Prints line to '/etc/fstab' file. """
