diff -Nur a/src/modules/grubcfg/main.py b/src/modules/grubcfg/main.py
--- a/src/modules/grubcfg/main.py	2018-01-02 14:37:36.000000000 +0000
+++ b/src/modules/grubcfg/main.py	2019-01-26 22:47:25.005850844 +0000
@@ -44,16 +44,20 @@
     dracut_bin = libcalamares.utils.target_env_call(
         ["sh", "-c", "which dracut"]
         )
-    have_dracut = dracut_bin == 0  # Shell exit value 0 means success
+    plymouth_bin = libcalamares.utils.target_env_call(
+        ["sh", "-c", "which plymouth"]
+        )
+    # Shell exit value 0 means success
+    have_dracut = dracut_bin == 0
+    have_plymouth = plymouth_bin == 0
 
     use_splash = ""
     swap_uuid = ""
     swap_outer_uuid = ""
     swap_outer_mappername = None
 
-    if libcalamares.globalstorage.contains("hasPlymouth"):
-        if libcalamares.globalstorage.value("hasPlymouth"):
-            use_splash = "splash"
+    if have_plymouth: 
+        use_splash = "splash"
 
     cryptdevice_params = []
 
