diff -Nur a/etc/rc.conf b/etc/rc.conf
--- a/etc/rc.conf	2017-10-10 19:38:12.000000000 +0100
+++ b/etc/rc.conf	2017-10-22 09:24:59.365825762 +0100
@@ -196,7 +196,7 @@
 # cgroups version 1 on /sys/fs/cgroup.
 # "legacy" mounts cgroups version 1 on /sys/fs/cgroup
 # "unified" mounts cgroups version 2 on /sys/fs/cgroup
-#rc_cgroup_mode="hybrid"
+rc_cgroup_mode="legacy"
 
 # This is a list of controllers which should be enabled for cgroups version 2.
 # If hybrid mode is being used, controllers listed here will not be
