diff -Nur a/init-fbcondecor b/init-fbcondecor
--- a/init-fbcondecor	2011-06-12 22:45:59.000000000 +0100
+++ b/init-fbcondecor	2017-08-14 16:47:22.051031795 +0100
@@ -1,10 +1,10 @@
-#!/sbin/runscript
+#!/sbin/openrc-run
 # Copyright 1999-2007 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
 depend() {
-	before xdm
-	use localmount
+	after localmount
+	before dkms
 }
 
 start() {
