diff -Nur a/usr/src/virtualbox-6.1.18/vboxnetflt/linux/VBoxNetFlt-linux.c b/usr/src/virtualbox-6.1.18/vboxnetflt/linux/VBoxNetFlt-linux.c
--- a/usr/src/virtualbox-6.1.18/vboxnetflt/linux/VBoxNetFlt-linux.c	2021-01-07 15:41:28.000000000 +0000
+++ b/usr/src/virtualbox-6.1.18/vboxnetflt/linux/VBoxNetFlt-linux.c	2021-03-16 16:27:00.179818747 +0000
@@ -69,6 +69,7 @@
 #include <iprt/mp.h>
 #include <iprt/mem.h>
 #include <iprt/time.h>
+#include <linux/ethtool.h>
 
 #define VBOXNETFLT_OS_SPECFIC 1
 #include "../VBoxNetFltInternal.h"
